bindings/python3/econf.py

```diff
@@ -0,0 +1,996 @@
+    :return: value of the key
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_result = c_uint64()
+    err = LIBECONF.econf_getUInt64Value(ef._ptr, group, c_key, byref(c_result))
+    if err:
+        _exceptions(err, f"get_uint64_value failed with error: {err_string(err)}")
+    return c_result.value
+
+
+def get_float_value(ef: EconfFile, group: str, key: str) -> float:
+    """
+    Return a float value for given group/key
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :return: value of the key
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_result = c_double()
+    err = LIBECONF.econf_getDoubleValue(ef._ptr, group, c_key, byref(c_result))
+    if err:
+        _exceptions(err, f"get_double_value failed with error: {err_string(err)}")
+    return c_result.value
+
+
+def get_string_value(ef: EconfFile, group: str, key: str) -> str:
+    """
+    Return a string value for given group/key
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :return: value of the key
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_result = c_char_p()
+    err = LIBECONF.econf_getStringValue(ef._ptr, group, c_key, byref(c_result))
+    if err:
+        _exceptions(err, f"get_string_value failed with error: {err_string(err)}")
+    return c_result.value.decode("utf-8")
+
+
+def get_bool_value(ef: EconfFile, group: str, key: str) -> bool:
+    """
+    Return a boolean value for given group/key
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :return: value of the key
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_result = c_bool()
+    err = LIBECONF.econf_getBoolValue(ef._ptr, group, c_key, byref(c_result))
+    if err:
+        _exceptions(err, f"get_bool_value failed with error: {err_string(err)}")
+    return c_result.value
+
+
+def get_int_value_def(ef: EconfFile, group: str, key: str, default: int) -> int:
+    """
+    Return an integer value for given group/key or return a default value if key is not found
+
+    :param ef: Key-Value storage object
+    :param group: desired group
+    :param key: key of the value that is requested
+    :param default: value to be returned if no key is found
+    :return: value of the key
+    """
+    if group:
+        group = _encode_str(group)
+    c_key = _encode_str(key)
+    c_result = c_int64()
+    c_default = _ensure_valid_int(default)
+    err = LIBECONF.econf_getInt64ValueDef(
+        ef._ptr, group, c_key, byref(c_result), c_default
+    )
+    if err:
+        if Econf_err(err) == Econf_err.ECONF_NOKEY:
+            return c_default.value
```

Comment:
that seems wrong:

(not tested)

```python
if err && EconfError(err) != EconfError.NOKEY:
   raise ECONF_EXCEPTION[err]("...")
return c_result.value
```

The default value should be transferred in the c_result variable

(the same issue for all the *_def functions)